### PR TITLE
Add arm64 preferences and test coverage

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

This PR introduces the following `arm64` specific preferences:

* `fedora.arm64`
* `rhel.9.arm64`

Test coverage is also introduced for using the following preferences with `arm64` `VirtualMachines` as there are no changes required:

* `ubuntu`
* `opensuse.leap`
* `centos.stream9`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
* `fedora.arm64` and `rhel.9.arm64` preferences have been introduced
* `arm64` test coverage has been extended for the `centos.stream9`,  `ubuntu` and  `opensuse.leap` preferences
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
IMO we should outline that if not otherwise stated `common-instancetypes` are implicitly x86? WDYT?

File: preferences/fedora/x86_64/metadata/metadata.yaml [hunk 1]
```diff
@@ -7,3 +7,5 @@ metadata:
     tags: "hidden,kubevirt,fedora"
     iconClass: "icon-fedora"
     openshift.io/display-name: "Fedora"
+  labels:
+    instancetype.kubevirt.io/arch: "x86_64"
```
Comment (0xFelix) [inline preferences/fedora/x86_64/metadata/metadata.yaml:11]: Should we add this to every preference?

Comment (lyarwood) [reply preferences/fedora/x86_64/metadata/metadata.yaml:11]: Yup, this is just a minimal draft only touching the Fedora preference.

**Comment by lyarwood** [comment]
> IMO we should outline that if not otherwise stated `common-instancetypes` are implicitly x86? WDYT?

Yup I'll document that but most of the instance types should be generic enough to work for both.

**Comment by 0xFelix** [comment]
@lyarwood Right, I meant the preferences of common-instancetypes.

File: preferences/fedora/amd64/metadata/metadata.yaml [hunk 2]
```diff
@@ -6,4 +6,6 @@ metadata:
   annotations:
     tags: "hidden,kubevirt,fedora"
     iconClass: "icon-fedora"
-    openshift.io/display-name: "Fedora"
+    openshift.io/display-name: "Fedora (x84_64)"
+  labels:
+    instancetype.kubevirt.io/arch: "amd64"
```
Comment (0xFelix) [inline preferences/fedora/amd64/metadata/metadata.yaml:11]: nit: Add newline at end of file

File: preferences/fedora/arm64/metadata/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: instancetype.kubevirt.io/v1beta1
+kind: VirtualMachinePreference
+metadata:
+  name: metadata
+  annotations:
+    tags: "hidden,kubevirt,fedora"
+    iconClass: "icon-fedora"
+    openshift.io/display-name: "Fedora (aarch64)"
```
Comment (0xFelix) [inline preferences/fedora/arm64/metadata/metadata.yaml:9]: What's better to use as display name? aarch64 or arm64?

File: preferences/fedora/kustomization.yaml [hunk 4]
```diff
@@ -3,26 +3,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
-      - op: replace
-        path: /metadata/name
-        value: fedora
-  - target:
-      kind: VirtualMachineClusterPreference
-    patch: |-
-      - op: replace
-        path: /metadata/name
-        value: fedora
+  - ./arm64
```
Comment (0xFelix) [inline preferences/fedora/kustomization.yaml:6]:
```suggestion
  - ./aarch64
```

Use the equivalent to `amd64`?

File: preferences/fedora/arm64/metadata/metadata.yaml (hunk 3 above)
Comment (lyarwood) [reply preferences/fedora/arm64/metadata/metadata.yaml:9]: `aarch64` is fine for the display name IMHO.

File: tests/functests/instancetype_test.go [hunk 5]
```diff
@@ -106,27 +114,31 @@ var _ = Describe("Common instance types func tests", func() {
...
 				[]testFn{expectGuestAgentToBeConnected, expectSSHToRunCommandOnLinux("fedora")}),
-			Entry("[test_id:10740] CentOS 7", centos7ContainerDisk, "centos.7",
+			Entry("[test_id:10740] CentOS 7", centos7ContainerDisk, map[string]string{"amd64": "centos.7"},
 				[]testFn{expectGuestAgentToBeConnected, expectSSHToRunCommandOnLinux("centos")}),
-			Entry("[test_id:10744] CentOS Stream 8", centosStream8ContainerDisk, "centos.stream8",
+			Entry("[test_id:10744] CentOS Stream 8", centosStream8ContainerDisk, map[string]string{"amd64": "centos.stream8"},
 				[]testFn{expectGuestAgentToBeConnected, expectSSHToRunCommandOnLinux("centos")}),
-			Entry("[test_id:10745] CentOS Stream 9", centosStream9ContainerDisk, "centos.stream9",
+			Entry("[test_id:10745] CentOS Stream 9", centosStream9ContainerDisk,
+				map[string]string{"amd64": "centos.stream9", "arm64": "centos.stream8"},
```
Comment (0xFelix) [inline tests/functests/instancetype_test.go:124]:
```suggestion
				map[string]string{"amd64": "centos.stream9", "arm64": "centos.stream9"},
```

Wrong pref used for arm?

**Review by 0xFelix — APPROVED** [review]
/approve
/lgtm

Thanks!

**Pull request merged**